### PR TITLE
Highlight incompatibility with Adafruit ecosystem

### PR DESCRIPTION
**Pull request by dglaude**
I try to manage expectation about this board and avoid short circuit or broken sensor. 1) I almost burned myself by mixing LOLIN I2C JST SH accessories and Adafruit QTPy, reverse way, I could have destroyed an Adafruit sensor by connecting it to Lolin S2 Pico. 2) The screen does not work as expected for a build-in screen, DISPLAY is not defined when you `import board`

I did hesitate between making a special section like "Caveats" or highlighting how it deviate from one can expect from a board running CircuitPython.

### CODE REVIEW

**Comment by deshipu** [comment]
I suppose the display initialization could be added to board.c for this board?

**Comment by dglaude** [comment]
Hi @deshipu , yes there must be a way to have the screen work as expected but apparently it is the first board where the screen is connected in I2C so there was no example to copy from.
https://github.com/adafruit/circuitpython/issues/7325#issuecomment-1345048637

If this PR make's it, and then later someone implement that the proper way, we can remove the warning (or say starting with CP9.0.0-alpha1 this statement is not true anymore).

**Comment by evildave666** [comment]
To be fair, the S2 Pico circuitpython-org entry (I wrote it) does have a mention that the connector is not Stemma-QT pinout.

File: _board/lolin_s3.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ WiFi & Bluetooth 5 (LE) development boards based ESP32-S3-WROOM-1.
   - `EN` RESET button
   - `GPIO0` BOOT button
   - `GPIO38` WS2812B RGB LED 
-  - Lolin I2C JST SH 4-pin port (not QWIIC/Stemma-Qt pinout) using `GPIO42` (SDA) and `GPIO41` (SCL)
+  - Lolin I2C JST SH 4-pin port (**not QWIIC/Stemma-Qt pinout**) using `GPIO42` (SDA) and `GPIO41` (SCL)
```
Comment (dhalbert) [inline _board/lolin_s3.md:35]:
```suggestion
  - Lolin I2C JST SH 4-pin port (**does not match QWIIC/Stemma-Qt pinout**) using `GPIO42` (SDA) and `GPIO41` (SCL)
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
Thanks! One suggestion for increased clarity.

**Comment by dglaude** [comment]
There might be a problem with the CI and Jekyll in particular, not sure why those simple change break it.

**Review by jepler — APPROVED** [review]
Approving based on Dan's previous review

**Pull request merged**